### PR TITLE
scmi-msg drivers: various fixes

### PR DESCRIPTION
**Pull request by etienne-lms**

### CODE REVIEW

File: core/drivers/scmi-msg/clock.c [hunk 1]
```diff
@@ -303,6 +303,12 @@ static void scmi_clock_describe_rates(struct scmi_msg *msg)
 		status =  plat_scmi_clock_rates_array(msg->agent_id, clock_id,
 						      plat_rates, &ret_nb);
 		if (status == SCMI_SUCCESS) {
+			assert((sizeof(p2a) % sizeof(uint32_t)) == 0U);
```
Comment (jforissier) [inline core/drivers/scmi-msg/clock.c:306]: `COMPILE_TIME_ASSERT()`?

File: core/drivers/scmi-msg/clock.c [hunk 2]
```diff
@@ -303,6 +303,12 @@ static void scmi_clock_describe_rates(struct scmi_msg *msg)
...
+			if (((size_t)msg->out % __alignof__(uint32_t)) != 0U) {
```
Comment (jforissier) [inline core/drivers/scmi-msg/clock.c:307]: The cast to `(size_t)` looks weird, is it needed? Would `(uintptr_t)` be better?
`!= 0U` is not really needed.

Comment (etienne-lms) [reply core/drivers/scmi-msg/clock.c:307]: Actually i this change seems useless. the rate array, in the buffer referred by `msg->out` is always 32bit alignment by SCMI message layout definition. So this commit "core: scmi-drv: check buffer alignment on clock rate array" could be discarded.

File: core/drivers/scmi-msg/clock.c (hunk 1 above)
Comment (etienne-lms) [reply core/drivers/scmi-msg/clock.c:306]: expected true by design of the SCMI spec.
Failure on alignment is expected to catch unaligned accesses.
If `CFG_SCTLR_ALIGNMENT_CHECK=y`.

**Comment by etienne-lms** [comment]
Commit "core: scmi-drv: check buffer alignment on clock rate array"  abandoned, not needed.

Tag applied on the 2 reviewed patches.
Rebased.

**Pull request merged**